File: CSharp/Containers/MongoDB-REST/MongoDB.REST/Controllers/MongoDB_REST_Controller.cs
using Microsoft.AspNetCore.Mvc;
using MongoDB.REST.Client;
using MongoDB.CRUD.Client;
using System.Text.Json;

namespace MongoDB.REST.Controllers;

[ApiController]
[Route("api/mongodb")]
public class MongoDB_REST_Controller : ControllerBase
{
    static private IActionResult ToActionResult(REST_Response response)
    {
        return new ContentResult
        {
            StatusCode = response.StatusCode,
            ContentType = response.ContentType,
            Content = response.Body
        };
    }

    private readonly IMongoDB_Client_Builder _mongoDB_Client_Builder;

    public MongoDB_REST_Controller(IMongoDB_Client_Builder mongoDB_Client_Builder)
    {
        _mongoDB_Client_Builder = mongoDB_Client_Builder;
    }

    private MongoDB_REST_Client CreateMongoDBClient()
    {
        return new MongoDB_REST_Client(new MongoDB_CRUD_Client(_mongoDB_Client_Builder));
    }

    [HttpGet("{db_name}/{collection_name}/{id}")]
    public async Task<IActionResult> Get(string db_name, string collection_name, string id)
    {
        return ToActionResult(await CreateMongoDBClient().GetAsync(db_name, collection_name, id));
    }

    [HttpPost("{db_name}/{collection_name}")]
    public async Task<IActionResult> Post(string db_name, string collection_name, [FromBody] object body)
    {
        return ToActionResult(await CreateMongoDBClient().PostAsync(db_name, collection_name, JsonSerializer.Serialize(body)));
    }

    [HttpPut("{db_name}/{collection_name}/{id}")]
    [HttpPatch("{db_name}/{collection_name}/{id}")]
    public async Task<IActionResult> Put(string db_name, string collection_name, string id, [FromBody] object body)
    {
        return ToActionResult(await CreateMongoDBClient().PutAsync(db_name, collection_name, id, JsonSerializer.Serialize(body)));
    }

    [HttpDelete("{db_name}/{collection_name}/{id}")]
    public async Task<IActionResult> Delete(string db_name, string collection_name, string id)
    {
        return ToActionResult(await CreateMongoDBClient().DeleteAsync(db_name, collection_name, id));
    }
}

File: Examples/TypeScript/Containers/NodeJS-MongoDB-REST/src/controllers/mongodb-rest.controller.ts
import { Request, Response } from 'express';
import MongoDB_CRUD_Client from '../services/MongoDB/CRUD/mongodb-crud-client';
import MongoDB_REST_Client from '../services/MongoDB/REST/mongodb-rest-client';

export class MongoDB_REST_Controller {
    constructor() {
        this.get = this.get.bind(this);
        this.post = this.post.bind(this);
        this.put = this.put.bind(this);
        this.delete = this.delete.bind(this);
    }

    private createMongoDBClient(): MongoDB_REST_Client {
        return new MongoDB_REST_Client(new MongoDB_CRUD_Client());
    }

    async get(req: Request, res: Response) {
        const client = this.createMongoDBClient();
        const response = await client.get(req.params.db_name as string, req.params.collection_name as string, req.params.id as string);
        res.status(response.status_code).type(response.content_type).send(response.body);
    }

    async post(req: Request, res: Response) {
        const client = this.createMongoDBClient();
        const response = await client.post(req.params.db_name, req.params.collection_name, req.body);
        res.status(response.status_code).type(response.content_type).send(response.body)
    }

    async put(req: Request, res: Response) {
        const client = this.createMongoDBClient();
        const response = await client.put(req.params.db_name as string, req.params.collection_name as string, req.params.id as string, req.body);
        res.status(response.status_code).type(response.content_type).send(response.body);
    }

    async delete(req: Request, res: Response) {
        const client = this.createMongoDBClient();
        const response = await client.delete(req.params.db_name, req.params.collection_name, req.params.id);
        res.status(response.status_code).type(response.content_type).send(response.body)
    }
}